src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/PackageReferenceCommands/ListPackage/ListPackageCommandRunner.cs

```diff
@@ -181,19 +197,21 @@ private static void WarnForHttpSources(ListPackageArgs listPackageArgs)
             {
                 if (httpPackageSources.Count == 1)
                 {
-                    listPackageArgs.Logger.LogWarning(
+                    projectModel.AddProjectInformation(
```

Comment:
Shouldn't this be something logged in the warnings of the model? I know it's not necessarily list package specific, but might simplify the logic?

I'm on the  fence about this. It's slightly weird to have it in the json output.

Reply:
> Shouldn't this be something logged in the warnings of the model? I know it's not necessarily list package specific, but might simplify the logic?
> 

Currently this same `http` warning printed in each project model, it's existing behavior. Do you suggest make it print only once for solution? Sources are only printed somehow `http warning` is printed repeatedly for each project.
Could you elaborate on make it simple?
![image](https://user-images.githubusercontent.com/8766776/199125214-f9aa1830-8b83-4719-a887-888f97bc91e6.png)

> I'm on the fence about this. It's slightly weird to have it in the json output.

If someone using this report detect warning/errors then skipping only 1 doesn't make sense. Maybe using `http connection` could be against the corp/government mandated compliance, that case customer wouldn't notice it.

Reply:
> Currently this same http warning printed in each project model, it's existing behavior. Do you suggest make it print only once for solution? Sources are only printed somehow http warning is printed repeatedly for each project.

Nope. Per project is ok.

I saw LoggerWarning, why that instead of just Warning? 

Why I'm saying it's make it simple is because this is the only usage of LoggerWarning.

Reply:
> 
> I saw LoggerWarning, why that instead of just Warning?
> 

No other warning has this `warn` prefix, only this one has this. Without this extra enum `LoggerWarning` I have to check index of `You are running the 'list package' operation with an 'HTTP' source` in warning message to detect if it's `http warning` and prefix with `warn : ` before print for console, but with this one we don't have to do checking that hard coded string check. I feel it's cleaner than other option.

![image](https://user-images.githubusercontent.com/8766776/199129128-64b999f5-e9bb-440d-98eb-dd082715710e.png)

Reply:
Are  there any other warnings today?

Everytihng that goes through the logger will have that prefix.

Reply:
Yes, there is another type of warning which doesn't have `warn : ` in front.
https://github.com/NuGet/NuGet.Client/blob/2d98be1ae155616ca080955fe205e01574f74134/src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/PackageReferenceCommands/ListPackage/ListPackageCommand.cs#L205-L212
![image](https://user-images.githubusercontent.com/8766776/199146875-4df72538-2300-4ec2-ba2f-4fbd0846a56d.png)

Reply:
I see, so this is more of a log message than a warning.

At least that's what it was previously.

Reply:
I think similar feedback still applies. 
That's a function of the options/input to the command. It doesn't have to be part of the model/analysis of a project.

Reply:
> I see, so this is more of an information message than a warning.
> 
> At least that's what it was previously.

For me it's warning about incompatible option combination, so prevents customer making wrong assumption about report, so I believe this one is actually a warning, please note we designed it would go into `Problems` sections in spec. I'll remove the `ProblemType.Information` type we need to keep this warning type, also removing this from model/analysis means I need to move logic into both console and json `Renderer` that means duplication of code.
Instead, I can remove ``ProblemType.LoggerWarning` and manually check if it's `Http warning` then add `warn : ` in front.

Reply:
> For me it's warning about incompatible option combination, so prevents customer making wrong assumption about report, so I believe this one is actually a warning, please note we designed it would go into Problems sections in spec.

That's fair. 

> I'll remove the ProblemType.Information type we need to keep this warning type, also removing this from model/analysis means I need to move logic into both console and json Renderer that means duplication of code.

I think it's about which pattern you lean into. If we want to have a complete separation between the data generated and we end up with duplication, which could just be just a helper.

> Instead, I can remove ProblemType.LoggerWarning and manually check if it's Http warning then add warn : in front.

I think it'd be great if we have only one warning type.

Reply:
> > Instead, I can remove ProblemType.LoggerWarning and manually check if it's Http warning then add warn : in front.
> 
> I think it'd be great if we have only one warning type.

I removed `ProblemType.LoggerWarning`, now we have only 1 warning type `ProblemType.Warning`.